docs_src/design/adr/system-management/0006-EdgeX-Metrics.md

```diff
@@ -0,0 +1,93 @@
+- Avg request size
+- Max request size
+- Auth/auth failures (once we get there)
+
+**Application Services**
+- Processing time for a pipeline
+- DB access times
+- How often are we failing export to be sent to db to be retried at a later time
+- What is the current queue Size
+- How much data is being sent to an endpoint(Volume)
+- Track important expected/Common Error Counts and Stats
+- Number of invalid Events seen
+- Number of events processed
+
+**Security**
+
+Security metrics may be more difficult to ascertain as they are cross service metrics (perhaps put in the general category).  Also, true threat detection based on metrics may be a feature best provided by 3rd party based on particular threats and security profile needs.
+- Number of logins and login failures per service and within a given time
+- Number of secrets accessed per service name 
+- Count of any accesses and failures to the data persistence layer
+- Count of start and restart attempts 
+- Registration of new devices and services (allowing for future quarantine mechanism) 
+- Indication of any new uploading of new CA/import to vault
+- 
+**Device Services**
+- Number of devices managed by this DS
+- Device Requests (which may be more informative that reading counts and rates)
+- Cumulative number succeeded / failed
+  - Number processed in last 1 / 5 / 15 minutes (or other defined intervals)
+  - Average time to process: all-time / last 5
+
+**Miscellaneous requirements**
+
+- It would be nice to be able to set granularity of telemetry based on INFO, DEBUG, VERBOSE; something like a telemetryLevel option in config
+- Current system management agent metric endpoint doesn’t return metrics in a timely fashion; i.e. takes more than a second to return. Talking directly to an individual service for metrics is quicker.
+- Metrics that system management agent provides shouldn’t include those that aren’t EdgeX-specific (i.e. memory and cpu usage) and which can’t be gathered using other tools/technologies. events. Based on prior experience, don’t instrument all the things. Define your metrics first and provide an initial limited view. 
+- Need to avoid service bloat.  EdgeX is not an enterprise system.  How can we implement in a concise and economical way.
+
+
+## Context 
+It is recommended that each service be responsible for its own metric data collection and deposit into the central data repository – core data.  Core data is already uniquely qualified to receive service data “readings”, persist them and make them available to application services for other apps, enterprise, cloud system use or internal decision making (rules engine).
+
+Metadata descriptions along with typing/tagging value descriptors (or their equivalent) would define control plane data from each service.  In terms of data, control plane data would look no different than sensor data.  The difference is that the data could come from every service (versus device services).
```

Comment:
While I agree that, as state here (`"In terms of data, control plane data would look no different than sensor data..."`), I think we should not miss underscoring that there are two messaging subsystems, along the lines of [the Control Bus design pattern](https://www.enterpriseintegrationpatterns.com/patterns/messaging/ControlBus.html).